datatree/io.py

```diff
@@ -68,16 +68,8 @@ def _open_datatree_netcdf(filename: str, **kwargs) -> DataTree:
         tree_root = DataTree.from_dict({"/": ds})
         for path in _iter_nc_groups(ncds):
             subgroup_ds = open_dataset(filename, group=path, **kwargs)
+            tree_root[path] = DataTree(name=path, data=subgroup_ds)
 
-            # TODO refactor to use __setitem__ once creation of new nodes by assigning Dataset works again
```

Comment:
This comment was actually referring to the difference between

`dt['folder'] = DataTree(data=ds)`
and
`dt['folder'] = ds`,

the latter of which currently doesn't work, but also perhaps doesn't need to.